src/less.rs

```diff
@@ -3,21 +3,32 @@
 use std::ffi::OsStr;
 use std::process::Command;
 
-pub fn retrieve_less_version(less_path: &dyn AsRef<OsStr>) -> Option<usize> {
+#[derive(Debug, PartialEq)]
+pub enum LessVersion {
+    Less(usize),
+    BusyBox,
+}
+
+pub fn retrieve_less_version(less_path: &dyn AsRef<OsStr>) -> Option<LessVersion> {
     let resolved_path = grep_cli::resolve_binary(less_path.as_ref()).ok()?;
 
     let cmd = Command::new(resolved_path).arg("--version").output().ok()?;
-    parse_less_version(&cmd.stdout)
+    if cmd.status.success() {
+        parse_less_version(&cmd.stdout)
+    } else {
+        parse_less_version(&cmd.stderr)
+    }
 }
 
-fn parse_less_version(output: &[u8]) -> Option<usize> {
-    if !output.starts_with(b"less ") {
-        return None;
+fn parse_less_version(output: &[u8]) -> Option<LessVersion> {
+    match std::str::from_utf8(output) {
```

Comment:
It would be good if we could continue being cautious about encoding, and check if this is less or BusyBox before assuming the output is UTF-8. Otherwise we are more or less certain to crash in scenarios where people use odd pagers (for whatever reason) that does not output UTF-8.